### ISSUE TITLE
Unpin puppetdb-termini

### ISSUE THREAD

**Issue by jvperrin** [ocf/puppet#956]
Currently, we've pinned the `puppetdb-termini` package to `6.4.0-1stretch` so that it works with octocatalog-diff: https://github.com/ocf/puppet/blob/4e37cbedd228d89bc2f32234dbb4fc54114faa9d/modules/ocf_ocfweb/manifests/dev_config.pp#L28-L33

However, we'll want to unpin it eventually so that we can get newer upgrades and so that it keeps working with the move to Puppet 7 (whenever that happens). We can't really stay pinned forever.

These are the kind of errors that it produces: https://gist.github.com/ocfbot/58ac942aaf361db56aedb5594b3faaad

Excerpt:
```
W, [2020-05-01T17:16:46.944698 #6391]  WARN -- : Puppet command failed: STDOUT:
STDERR:
  Error: Failed to initialize SSL: The CRL is missing from '/tmp/ocd-ipc-20200501-6308-6kr8np/ocd-builddir-20200501-6391-1kvspcg/var/ssl/crl.pem'
  Error: Run `puppet agent -t`
  Error: Cached facts for acid.ocf.berkeley.edu failed: Failed to find facts from PuppetDB at puppet:8140: The CRL is missing from '/tmp/ocd-ipc-20200501-6308-6kr8np/ocd-builddir-20200501-6391-1kvspcg/var/ssl/crl.pem'
  Error: Failed to initialize SSL: The CRL is missing from '/tmp/ocd-ipc-20200501-6308-6kr8np/ocd-builddir-20200501-6391-1kvspcg/var/ssl/crl.pem'
  Error: Run `puppet agent -t`
  Error: The CRL is missing from '/tmp/ocd-ipc-20200501-6308-6kr8np/ocd-builddir-20200501-6391-1kvspcg/var/ssl/crl.pem'
  Error: Failed when searching for node acid.ocf.berkeley.edu: Could not retrieve facts for acid.ocf.berkeley.edu: The CRL is missing from '/tmp/ocd-ipc-20200501-6308-6kr8np/ocd-builddir-20200501-6391-1kvspcg/var/ssl/crl.pem'
```

I believe this is due to https://github.com/puppetlabs/puppetdb/commit/8714ba9288da588e9fd197902332dd7c04635f10 since this fails with `puppetdb-termini` 6.5.0 and above but passes with 6.4.0, and this is one of three commits to change files included in the package (within [this directory](https://github.com/puppetlabs/puppetdb/tree/6.5.0/puppet/lib/puppet)) between 6.4.0 and 6.5.0. This is the only commit that obviously touches some SSL-related stuff, so it seems like the best candidate to investigate.

**Comment by jvperrin** [comment]
It looks like from the puppetdb logs (`/var/log/puppetlabs/puppetdb/puppetdb-access.log` in particular), that `octocatalog-diff` is able to make queries to get the list of nodes and get a list of their facts, but can't do more detailed resource lookups:

Failed run: https://i.fluffy.cc/FdjB7TzNZldnKlNFfv767ZvkHZqXJk3B.html
Successful run: https://i.fluffy.cc/lCkTFZGmDkxRNnlrzMrJHBt1bRdgQPJj.html

**Comment by jvperrin** [comment]
Looks like just providing an additional `--certificate_revocation=false` to the puppet commands used fixes this just fine. https://github.com/puppetlabs/puppetdb/commit/3055eb0ab544fceb74a019c2b5dcfbaa768eba7c in 6.6.0 gives some better handling around using SSL in masterless settings like this.

Anyway, with that option added to [our `octocatalog-diff` fork](https://github.com/ocf/octocatalog-diff) should make it all work fine. I've also created https://github.com/ocf/octocatalog-diff/issues/1 for unforking that repo (contributing it back upstream) as it seems it's still maintained and there's some other people that would find it useful.

**Comment by jvperrin** [comment]
Closed by #957 (and upgrading `puppetdb-termini`/`octocatalog-diff` across all hosts)
